### PR TITLE
ENH: stats.bartlett: add native `axis` and array API support

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Toward gh-20544

#### What does this implement/fix?
Adds native support for `axis` and adds array API support to `stats.bartlett`.

### CODE REVIEW

File: scipy/stats/_morestats.py [hunk 1]
```diff
@@ -3051,30 +3052,41 @@ def bartlett(*samples):
...
     k = len(samples)
     if k < 2:
         raise ValueError("Must enter at least two input sample vectors.")
 
-    # Handle empty input and input that is not 1d
+    # Handle 1d empty input; _axis_nan_policy takes care of N-D
     for sample in samples:
-        if np.asanyarray(sample).size == 0:
-            NaN = _get_nan(*samples)  # get NaN of result_dtype of all samples
+        if xp.asarray(sample).size == 0:
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:3063]:
```suggestion
        if xp_size(xp.asarray(sample)) == 0:
```

File: scipy/stats/tests/test_morestats.py [hunk 2]
```diff
@@ -735,33 +735,35 @@ def test_alternative_approx(self):
         assert_allclose(pval_l, 1-pval/2, atol=1e-12)
 
 
+@array_api_compatible
```
Comment (mdhaber) [inline scipy/stats/tests/test_morestats.py:738]:
```suggestion
@array_api_compatible
@pytest.mark.usefixtures("skip_xp_backends")
@pytest.mark.skip_xp_backends(cpu_only=True, reasons=['Uses NumPy for pvalue'])
```
This skips the test for CuPy, which isn't running on CI (so the failure is not visible).

File: scipy/stats/_morestats.py [hunk 3]
```diff
@@ -2841,7 +2842,7 @@ def ansari(x, y, alternative='two-sided'):
 
 
 @_axis_nan_policy_factory(BartlettResult, n_samples=None)
-def bartlett(*samples):
+def bartlett(*samples, axis=0):
```
Comment (tupui) [inline scipy/stats/_morestats.py:2845]: Should be keyword only no?

**Review by tupui — APPROVED** [review]
LGTM, just a question before getting in.

File: scipy/stats/_morestats.py (hunk 3 above)
Comment (mdhaber) [reply scipy/stats/_morestats.py:2845]: Yes. It is.

Comment (tupui) [reply scipy/stats/_morestats.py:2845]: Ah yes 👍 always forget the distinction with the splat for any number of positional args. Sorry for the noise

**Comment by mdhaber** [comment]
Thanks @tupui!

**Pull request merged**